arch/arm/src/imx9/imx9_flexcan.c

```diff
@@ -811,7 +812,7 @@ static int imx9_txpoll(struct net_driver_s *dev)
    * the field d_len is set to a value > 0.
    */
 
-  flags = spin_lock_irqsave(NULL);
+  flags = spin_lock_irqsave(&priv->lock);
```

Comment:
sched_lock

Reply:
But I didn't find any calls to sem_post in the protected code.

Reply:
https://github.com/apache/nuttx/blob/master/arch/arm/src/imx9/imx9_flexcan.c#L645

Reply:
done